### PR TITLE
ju/ednx/BC-19 Add eox-core and create dependencies folder structure

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
This add the dependencies' structure for edunext requirements and includes the first dependency(eox-core)

### CODE REVIEW

File: requirements/edunext/base.in [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#
+# DON'T JUST ADD NEW DEPENDENCIES!!!
+#
+# If you open a pull request that adds a new dependency, you should:
+#   * verify that the dependency has a license compatible with AGPLv3
+#   * confirm that it has no system requirements beyond what we already install
+#   * run "make upgrade" to update the detailed requirements files
+#
+
+-c ../constraints.txt
```
Comment (morenol) [inline requirements/edunext/base.in:11]: @andrey-canon 

What do you think about something like 

```-c ../edx/base.txt```

So we don't have to add the constraints for those packages?

File: requirements/edunext/base.in [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+-r github.in              # Forks and other dependencies not yet on PyPI
+
+# Please follow these guidelines whenever you change this file:
+#
+# 1. When adding a new dependency, add a comment to the end of the line
```
Comment (felipemontoya) [inline requirements/edunext/base.in:17]: If we are doing this, please do this on eox-core ;)

**Review by felipemontoya — CHANGES_REQUESTED** [review]
I like the overall aproach. I would still like to add the comments on why we are using what we are using.

If this is well documented, do you think we still would find the dashboard table with the dependencies useful?

File: requirements/edunext/base.in [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#    an already-pinned package to its latest release, remove the version pin.
+# 3. Do not add indirect dependencies unless a version constraint is needed to
+#    avoid versions which have known problems.
+# 4. If the package is not needed in production, add it to another file such
+#    as development.in or testing.in instead.
+
+###################
+# eduNEXT plugins #
+###################
+eox-core                            # Edunext core modifications. This plugin is necessary since this contains the most of edunext custom changes and the edunext logic businesses[required by saas].
```
Comment (morenol) [inline requirements/edunext/base.in:29]: In master, we are using eox-core[sentry,tpa].
Will that change be part of a future PR? or should we add that in this PR?

Comment (andrey-canon) [reply requirements/edunext/base.in]: done

**Review by morenol — APPROVED** [review]

**Pull request merged**